script/zip-comp-logs

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+"""
+A script to zip the logs produced by a series of competition matches.
+"""
+
+import re
+import shutil
+import argparse
+import datetime
+import tempfile
+from pathlib import Path
+
+
+def zip_folder(folder: Path, args: argparse.Namespace) -> None:
+    tla = folder.name
+    zip_name = f'team-{tla}'
+
+    if args.suffix:
+        zip_name += f'-{args.suffix}'
+    if not args.no_date:
+        zip_name += f'-{datetime.date.today()}'
+
+    shutil.make_archive(
+        f'{args.output_dir / zip_name}',
+        'zip',
+        root_dir=folder,
+        base_dir='.',
+    )
+
+
+def is_team_logs_folder(folder: Path) -> bool:
+    # Options for detecting which folders are logs:
+    # 1. everthing except recordings, matches
+    # 2. folders w/ matching zip file names
+    # 3. regex of folder name
+    # 4. regex of folder contents names
+
+    # Using option 3
+    # folder names are three uppercase letters and an optional number
+    if re.match(r'[A-Z]{3}[0-9]*$', folder.name):
+        return True
+
+    return False
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'archives_dir',
+        help=(
+            "The directory containing folders of the teams' robot logs, "
+            "named for the teams' TLAs. This directory also contains the "
+            "teams' robot code, as Zip archives and the match recordings."
+        ),
+        type=Path,
+    )
+    parser.add_argument(
+        'output_dir',
+        help=(
+            "The directory to place the resulting log Zip archives. "
+            "This should be an empty directory."
+        ),
+        type=Path,
+    )
+    parser.add_argument(
+        '--suffix',
+        help=(
+            "Adds a suffix to the Zip archive name. "
+            "Archive names follow the format: team-<TLA>-<suffix>-<date>.zip"
+        ),
+        default="",
+    )
+    parser.add_argument(
+        '--no-date',
+        help=(
+            "Remove the date from the archive name."
+        ),
+        action='store_true',
+    )
+    parser.add_argument(
+        '--combined',
+        help=(
+            "Create a combined archive which contains all the other log archives. "
+            "This is achieved by creating an archive of the contents of the output directory."
+        ),
```

Comment:
I'm not sure I follow what this means?

Perhaps it would help to (on the PR) provide an example of what the tree inside such an archive would look like?

Reply:
This was because the combined zip was generated by zipping the entire output directory, not just the zips created while running the script. ef3d27c1459c681484f721e29d58cdd77bf18822 changes this to just zip the files produced by the script.

Reply:
The contents of the combined zip is then:
```console
$ unzip -l combined2.zip 
Archive:  combined2.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
      913  2021-01-23 09:15   team-ABC-2021-01-23.zip
      682  2021-01-23 09:15   team-DEF-2021-01-23.zip
---------                     -------
     1595                     2 files
```